test/foo/lib/config/test_configloader.py

```diff
@@ -0,0 +1,26 @@
+import pytest
```

Comment:
S3 part is untested, there is a stubber for boto2: https://botocore.amazonaws.com/v1/documentation/api/latest/reference/stubber.html

Reply:
Clearly not written in tdd, multiple things are untested:
- failure to load 'env' file
- wrong env
- regardless of the meaning, reloading config
- loading environment keys